mlir/lib/Driver/CompilerDriver.cpp

```diff
@@ -648,7 +648,7 @@ LogicalResult QuantumDriverMain(const CompilerOptions &options, CompilerOutput &
     catalyst::utils::LinesCount::ModuleOp(*op);
     output.isCheckpointFound = options.checkpointStage == "mlir";
 
-    bool enzymeRun = false;
+    bool enzymeRun = options.checkpointStage == "O2Opt";
```

Comment:
I think a comment would be useful here it is initialized with that value, otherwise it is a bit out of the blue.

Reply:
Added. Thank you,